bls/src/main/java/tech/pegasys/teku/bls/BLS.java

```diff
@@ -93,19 +96,29 @@ public static BLSSignature aggregate(List<BLSSignature> signatures) {
    * <p>The standard says to return INVALID, that is, false, if the list of public keys is empty.
    * See also discussion at https://github.com/ethereum/eth2.0-specs/issues/1713
    *
+   * <p>We also return false if any of the messages are duplicates.
```

Comment:
Is there a consensus that this is the expected behavior? Worried that this might lead to consensus failures.

Reply:
Yes. The standard-defined behaviour is to return "INVALID". The question for us is whether that means return false or throw an exception (bacause it should only occur when there's a logical error somewhere in the code). I've chosen to return false here, but am open to either. In any case, it should not lead to consensus failure.

Reply:
👍 We're moving towards returning booleans from verifications instead of throwing exceptions (which is an anti-pattern as Adrian says), so the decision to return false here makes sense.